src/helm/proxy/clients/vertexai_client.py

```diff
@@ -113,3 +124,107 @@ def do_it():
             completions=completions,
             embedding=[],
         )
+
+
+class VertexAIChatClient(VertexAIClient):
+    """Client for Vertex AI chat models
+    This client is used for Gemini for example."""
+
+    def make_request(self, request: Request) -> RequestResult:
+        """Make a request"""
+        parameters = {
+            "temperature": request.temperature,
+            "max_output_tokens": request.max_tokens,
+            "top_k": request.top_k_per_token,
+            "top_p": request.top_p,
+            "stop_sequences": request.stop_sequences,
+            "candidate_count": request.num_completions,
+            # TODO #2084: Add support for these parameters.
+            # The parameters "echo", "frequency_penalty", and "presence_penalty" are supposed to be supported
+            # in an HTTP request (See https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/text),
+            # but they are not supported in the Python SDK:
+            # https://github.com/googleapis/python-aiplatform/blob/beae48f63e40ea171c3f1625164569e7311b8e5a/vertexai/language_models/_language_models.py#L968C1-L980C1
+            # "frequency_penalty": request.frequency_penalty,
+            # "presence_penalty": request.presence_penalty,
+            # "echo": request.echo_prompt,
+        }
+
+        completions: List[Sequence] = []
+
+        # TODO: support VLM such as "gemini-pro-vision"
+        model_name: str = ""
+        if request.model_engine == "gemini":
+            model_name = "gemini-pro"
```

Comment:
How does model versioning work? I believe sending a request with `gemini-pro` will just get the latest version.

Reply:
We will revisit later when Google releases more versions.